tests/main/snapctl-from-snap/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: Check that `snapctl` can be run from the snap
+
+prepare: |
+    snapbuild $TESTSLIB/snaps/snapctl-from-snap .
+    snap install --dangerous snapctl-from-snap_1.0_all.snap
+
+restore: |
+    rm snapctl-from-snap_1.0_all.snap
+
+execute: |
+    echo "Verify that context file exists and has proper permissions"
+    CONTEXT_FILE=/var/lib/snapd/contexts/snap.snapctl-from-snap
+    if ! test -f $CONTEXT_FILE ; then
+        echo "Context file $CONTEXT_FILE is missing"
+        exit 1
+    fi
+    if [ $(stat -c%a $CONTEXT_FILE) != "600" ]; then
+        echo "Incorrect permissions of file $CONTEXT_FILE"
+        exit 1
+    fi
+
+    echo "Verify that snapctl get can be executed by the app and shows the value set by configure hook"
+    OUTPUT=$(/snap/bin/snapctl-from-snap.app1)
```

Comment:
These test cases are both one-liners:
```
/snap/bin/snapctl-from-snap.app1 | MATCH bar
```
Given the comment above, it'll become get+set, but still simpler than what we have here.